portfolio/src/main/webapp/request.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * Models the requests sent to the algorithm for scheduling. This class is
+ * useful when the request information needs to be sent to the scheduling
+ * servlet through a POST request. This class contains all the request
+ * information that can be easily converted into a JSON string.
+ */
+class ScheduleRequest {
+  constructor(events, tasks, startTime, endTime) {
```

Comment:
Would be nice to include comments on each parameter's type in the constructor.

Reply:
Done.